Presentation.Web/app/components/it-contract/tabs/it-contract-tab-deadlines.controller.ts

```diff
@@ -12,20 +12,17 @@
                     localOptionServiceFactory.create(Kitos.Services.LocalOptions.LocalOptionType.TerminationDeadlineTypes).getAll()
                 ],
                 paymentMilestones: ["$http", "$stateParams", ($http, $stateParams) =>
-                    $http.get("api/paymentMilestone/" + $stateParams.id + "?contract=true").then(function (result) {
-                        return result.data.response;
-                    })],
+                    $http.get("api/paymentMilestone/" + $stateParams.id + "?contract=true").then(result => result.data.response)],
                 handoverTrialTypes: ["localOptionServiceFactory", (localOptionServiceFactory: Kitos.Services.LocalOptions.ILocalOptionServiceFactory) =>
                     localOptionServiceFactory.create(Kitos.Services.LocalOptions.LocalOptionType.HandoverTrialTypes).getAll()],
-                handoverTrials: ["$http", "$stateParams", ($http, $stateParams) => $http.get("api/handoverTrial/" + $stateParams.id + "?byContract=true").then(function (result) {
-                    return result.data.response;
-                })]
+                handoverTrials: ["$http", "$stateParams", ($http, $stateParams) => $http.get("api/handoverTrial/" + $stateParams.id + "?byContract=true").then(result => result.data.response)]
             }
         });
     }]);
 
-    app.controller("contract.DeadlinesCtrl", ["$scope", "$http", "$timeout", "$state", "$stateParams", "notify", "optionExtensions", "terminationDeadlines", "paymentMilestones", "handoverTrialTypes", "handoverTrials", "user", "moment", "$q",
-        function ($scope, $http, $timeout, $state, $stateParams, notify, optionExtensions, terminationDeadlines, paymentMilestones, handoverTrialTypes, handoverTrials, user, moment, $q) {
+    app.controller("contract.DeadlinesCtrl", ["$scope", "$http", "$timeout", "$state", "$stateParams", "notify", "optionExtensions", "terminationDeadlines", "paymentMilestones", "handoverTrialTypes", "handoverTrials", "user", "moment", "$q", "contract",
+        ($scope, $http, $timeout, $state, $stateParams, notify, optionExtensions, terminationDeadlines, paymentMilestones, handoverTrialTypes, handoverTrials, user, moment, $q, contract) => {
+            $scope.contract = contract;
             $scope.autosaveUrl = "api/itcontract/" + $scope.contract.id;
```

Comment:
Eftersom contract nu gives med som argument til controlleren vil det så ikke være bedre at bruge den her frem for at gå på $scope.contract? (Gælder flere steder i koden)

Reply:
jeg kan lige prøve og se om det giver bagslag men ja in controlleren burde det jo være det rigtige at gøre - specielt fordi den jo er i scope af alle underliggende funktioner